src/main/java/org/spongepowered/server/launch/plugin/PluginScanner.java

```diff
@@ -267,10 +278,32 @@ void scanJar(Path path, boolean classpath) {
                 }
             }
 
-            if (success && metadata == null) {
-                logger.warn("{} is missing a valid " + METADATA_FILE + " file. This is not a problem when testing plugins, however it is recommended "
-                        + "to include one in public plugins.\n"
-                        + "Please see https://docs.spongepowered.org/master/en/plugin/plugin-meta.html for details.", path);
+            if (success) {
+                if (metadata == null) {
+                    logger.warn("{} is missing a valid " + METADATA_FILE + " file."
```

Comment:
Since `METADATA_FILE` is a constant I was hoping the Java compiler would inline it... although since there are parameters used anyway it doesn't make much sense, yep.